samples/dnn/object_tracker.cpp

```diff
@@ -0,0 +1,355 @@
+}
+
+static int trackObject(const string& windowName, Ptr<Tracker> tracker, const string& inputName)
+{
+    namedWindow(windowName, WINDOW_AUTOSIZE);
+    FontFace fontFace("sans");
+    int stdSize = 20;
+    int stdWeight = 400;
+    int stdImgSize = 512;
+    int imgWidth = -1;
+    int fontSize = 50;
+    int fontWeight = 500;
+    Rect selectRect;
+
+    // Open a video file or an image file or a camera stream.
+    VideoCapture cap;
+
+    if (inputName.empty() || (isdigit(inputName[0]) && inputName.size() == 1))
+    {
+        int c = inputName.empty() ? 0 : inputName[0] - '0';
+        cout << "Trying to open camera #" << c << " ..." << endl;
+        if (!cap.open(c))
+        {
+            cout << "Capture from camera #" << c << " didn't work. Specify -i=<video> parameter to read from video file" << endl;
+            return 2;
+        }
+    }
+    else if (inputName.size())
+    {
+        string filePath = findFile(inputName);
+        if (!cap.open(filePath))
+        {
+            cout << "Could not open: " << inputName << endl;
+            return 2;
+        }
+    }
+
+    Mat image;
+
+    for (;;)
+    {
+        cap >> image;
+        if (image.empty())
+        {
+            cerr << "Can't capture frame. End of video stream?" << endl;
+            return 2;
+        }
+        else if (imgWidth == -1){
+            imgWidth = min(image.rows, image.cols);
+            fontSize = (stdSize*imgWidth)/stdImgSize;
+            fontWeight = (stdWeight*imgWidth)/stdImgSize;
+        }
+        const string label = "Press space bar to pause video to draw bounding box.";
+        Rect r = getTextSize(Size(), label, Point(), fontFace, fontSize, fontWeight);
+        r.height += 2 * fontSize; // padding
+        r.width += 10; // padding
+        rectangle(image, r, Scalar::all(255), FILLED);
+        putText(image, label, Point(10, fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+        putText(image, "Press space bar after selecting.", Point(10, 2*fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+        imshow(windowName, image);
+        int key = waitKey(200);
+        if (key == ' ')
+        {
+            selectRect = selectROI(windowName, image);
+            break;
+        }
+        if (key == 27) // ESC key to exit
+        {
+            exit(0);
+        }
+    }
+
+    Mat image_select = image.clone();
+
+    cout << "ROI=" << selectRect << endl;
+
+    tracker->init(image, selectRect);
+
+    TickMeter tickMeter;
+
+    for (int count = 0; ; ++count)
+    {
+        cap >> image;
+        if (image.empty())
+        {
+            cerr << "Can't capture frame " << count << ". End of video stream?" << endl;
+            break;
+        }
+
+        Rect rect;
+
+        tickMeter.start();
+        bool ok = tracker->update(image, rect);
+        tickMeter.stop();
+
+        float score = getTrackingScore(tracker);
```

Comment:
I propose to change cv::Tracker api and move getTrackingScore() method to base tracker class. -1 could be default value, if it's not supported by some implementation.
@vpisarev What do you think?